lib/commit-validator.js

```diff
@@ -1,31 +1,56 @@
+const lint = require('@commitlint/lint').default;
+
 class CommitValidator {
   constructor(options) {
-    this.patterns = [
-      new RegExp(
-        `^(${options.tasks.join('|')})\\((${options.scopes.join('|')})\\): [A-Z]{2,}-[0-9]+ .+`,
-      ),
-      new RegExp('^Revert ".+'),
-      new RegExp('^Releas(e|ing) v?\\d+\\.\\d+\\.\\d+'),
+    this.ignores = [
+      (commit) => new RegExp('^Releas(e|ing) v?\\d+\\.\\d+\\.\\d+').test(commit),
+      (commit) => new RegExp('^Revert ".+').test(commit),
+    ];
+
+    this.rules = {
+      'type-enum': [
+        2,
+        'always',
+        options.tasks
+          ? options.tasks
+          : ['feat', 'fix', 'docs', 'style', 'refactor', 'test', 'revert'],
+      ],
+      'scope-enum': [2, 'always', options.scopes ? options.scopes : []],
+      'subject-start-jira': [2, 'always'],
+    };
+
+    this.plugins = [
+      {
+        rules: {
+          'subject-start-jira': ({ subject }) => {
+            return [
+              subject ? subject.match(/^[A-Z]{2,}-\d+ /) : false,
+              `Your subject must contain a JIRA ticket (i.e. BIG-123).`,
+            ];
+          },
+        },
+      },
     ];
 
     this.errorMessage = this._getError(options.scopes, options.tasks);
   }
 
-  isValid(subject) {
-    return this.patterns.some((pattern) => pattern.test(subject));
+  async isValid(commit) {
+    return lint(commit, this.rules, {
+      defaultIgnores: false,
+      ignores: this.ignores,
+      plugins: this.plugins,
+    });
   }
 
-  validate(commitsToValidate) {
-    const commits = commitsToValidate.map((subject) => ({
-      valid: this.isValid(subject),
-      subject,
-    }));
+  async validate(commits) {
+    const results = await Promise.all(commits.map((commit) => this.isValid(commit)));
 
-    const valid = commits.every((commit) => commit.valid);
+    const valid = results.every((result) => result.valid);
     const message = valid ? 'All Good!' : this.errorMessage;
```

Comment:
I hate to keep this old error message here when we have commitlint ones, but wanted to keep the complexity of the PR down – keeping the old message for now.